### ISSUE TITLE
[Bug]: queryCoord should check the errorCode of GetRecoveryInfo

### ISSUE THREAD

**Issue by xige-16** [milvus-io/milvus#12147]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version:
- Deployment mode(standalone or cluster):
- SDK version(e.g. pymilvus v2.0.0rc2):
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

if rootCoord is not healthy, datacoord with return GetRecoveryInfo{ErrorCode: unexpectError} and err == nil to queryCoord, queryCoord should check the err and the ErrorCode together

**Comment by yanliang567** [comment]
@xige-16 is it done now

**Comment by xige-16** [comment]
the issue has been fixed by #12983
